Дз4

Yunnii opened the pull request:
Только мне кажется, с фуункциями sort для коллекций я полный хардкор устроила... может как-то по-аккуратнее можно?

Yunnii committed:
submodule qunit

Yunnii committed:
1 и 2 Части

Создан конструктор + методы для абстрактного объекта model
Класс Event унаследовался от Model
Написаны тесты на 1 и 2 пункт

Yunnii committed:
3 пункт выполнен

1. добавлены новые тесты, выделены в отделный каталог
2. создан конструктор + фильтр + фции для коллекции

Yunnii committed:
Почти доделано ^^

1. допилен 4 пункт
2. остался jslint + jsdoc

Yunnii committed:
Jslint

Yunnii committed:
Jsdoc

Yunnii committed:
маленький fix

azproduction commented on events.js:
```diff
@@ -0,0 +1,112 @@
+    return this.coming()
+               .filter(function (events) {
+            return events.start < now;
+        });
```
Вот так более читаемо

``` javascript
var result = this.coming()
    .filter(function (events) {
        return events.start < now;
    });

return result;
```

azproduction commented on events.js:
```diff
@@ -0,0 +1,112 @@
+﻿var Events = function (data) {
```
Ты, кстати, можешь использовать ~~паттерн модуль~~IEFE и избавиться от кучи `"use strict"`;

``` javascript
(function (exports) {
    "use strict";

    var Events = function (data) {
        Collection.apply(this, arguments);
    };

    inherits(Events, Collection);

    // your stuff

    exports.Events = Events;
}(window));
```

azproduction commented on events.js:
```diff
@@ -0,0 +1,112 @@
+Events.prototype.sortByTime = function (isAscending) {
+    "use strict";
+
+    if (isAscending || typeof isAscending === "undefined") {
```
лучше вместо такой длинной проверки сделать в начале вот так `isAscending = isAscending || false;` а потом просто `if (isAscending) {}`

azproduction commented on events.js:
```diff
@@ -0,0 +1,112 @@
+ * @param {isAscending} - необязательный параметр - указывает порядок сортировки. 
+ * при отсутствии сортируется по убыванию.
+ * @return коллекция объектов типа event
+*/
```
Формат JSDoc не корректный - http://code.google.com/p/jsdoc-toolkit/wiki/TagReference

```
description
%new_line%
@param {Type} name description
@param {Type} name description
%new_line%
@return {Type} description
```

``` javascript
/**
 * Создает новый экземпляр Circle по диаметру.
 *
 * @param {number} d Диаметр окружности.
 *
 * @return {Circle} Новый объект Circle.
 */
Circle.fromDiameter = function (d) {
    return new Circle(d / 2);
};
```

azproduction commented:
> Только мне кажется, с фуункциями sort для коллекций я полный хардкор устроила... может как-то по-аккуратнее можно?

про форматирование отписал

Yunnii commented:
в коллекции Events, когда вызывается, например фция

Events.prototype.byEndTime =  function () {

```
return this.sort(function (a, b) {
    return a.end - b.end;
});
```

};

идет вызов функции sort, которая возвращает объект типа Collection, это ведь не верно? должно Events возвращать?

Yunnii committed:
jslint

azproduction commented:
> должно Events возвращать?

Да, должно возвращать инстанс текущего конструктора. Можешь где-нибудь в прототипе сохранить ссылку "на себя" (`prototype.constructor = Events`), а в функции sort использовать.

azproduction commented on event.js:
```diff
@@ -0,0 +1,81 @@
+﻿((function (exports) {
```
Лишняя `(` скобка.

azproduction commented on events.js:
```diff
@@ -0,0 +1,122 @@
+    return this
+            .byRaiting()
+            .reverse();
+};
```
Не закрыта IEFE; Events и все подобные не экспортируются. `exports.Events = Events`;

azproduction commented:
Последний коммит все испортил. Я более чем уверен, что у тебя ничего не работает ;-)

Yunnii commented:
меня опечалило, что фции возвращают совсем не то, что надо и я даже тесты не прогнала. позор мне)

azproduction commented:
Вот так и познается вся мощь автоматических юнит-тестов!

azproduction commented:
Если интересно попробуй начать использовать Qunit PhantomJS Runner https://github.com/jquery/qunit/tree/master/addons/phantomjs с TravisCI ;-)

Yunnii committed:
Все )

1. Ошибули поправлены
2. Тесты работают
3. Фции возвращают для Collection - новую коллекцию типа Collection,
для Events - типа Events
